ttnn/cpp/ttnn/tensor/tensor.cpp

```diff
@@ -511,7 +511,24 @@ const tt::tt_metal::LegacyShape Tensor::strides() const { return tt::tt_metal::L
 
 uint32_t Tensor::volume() const { return tt::tt_metal::compute_volume(this->get_legacy_shape()); }
```

Comment:
Why not `return get_physical_shape().volume()` ?

Reply:
I think because `LegacyShape` can both represent padded or original shape. I think usages of `volume` should be switched over to `get_logical_volume` in most cases.

Reply:
I thought for LegacyShape you had to call without_padding() to get the non padded shape and compute_volume didn't do that.

Yes, it'd be good to check and update the usages of `volume`, although it sounds like that will be done in a separate PR:
```
These two calls are identical, except its better to ask tensor directly

mb call it physical volume?

Please let me do it in a separate PR
```

Reply:
It can be done, I am just scoping changes down for now.